src/pluto/pluto.py

```diff
@@ -525,69 +482,62 @@ def generate_graph(v1, v2):
         """
         )
 
-    create_mismatch(df_mismatch, v1, v2, v3, condo, mapped)
+    def create_expected(df, v1, v2):
 
-    create_null(df_null, v1, v2, v3, condo, mapped)
+        exp = df[df["v"].isin([v1, v2])]
 
-    create_aggregate(df_aggregate, v1, v2, v3, condo, mapped)
+        exp_records = exp.to_dict("records")
+        v1_exp = [i["expected"] for i in exp_records if i["v"] == v1][0]
+        v2_exp = [i["expected"] for i in exp_records if i["v"] == v2][0]
+        for field in [
+            "zonedist1",
+            "zonedist2",
+            "zonedist3",
+            "zonedist4",
+            "overlay1",
+            "overlay2",
+            "spdist1",
+            "spdist2",
+            "spdist3",
+            "ext",
+            "proxcode",
+            "irrlotcode",
+            "lottype",
+            "bsmtcode",
+            "bldgclasslanduse",
+        ]:
+            val1 = [i["values"] for i in v1_exp if i["field"] == field][0]
+            val2 = [i["values"] for i in v2_exp if i["field"] == field][0]
+            in1not2 = [i for i in val1 if i not in val2]
+            in2not1 = [i for i in val2 if i not in val1]
+            if len(in1not2) == 0 and len(in2not1) == 0:
+                pass
+            else:
+                st.markdown(f"### Expected value difference for {field}")
+                if len(in1not2) != 0:
+                    st.markdown(f"* in {v1} but not in {v2}:")
+                    st.write(in1not2)
+                if len(in2not1) != 0:
+                    st.markdown(f"* in {v2} but not in {v1}:")
+                    st.write(in2not1)
+
+    create_mismatch(data["df_mismatch"], v1, v2, v3, condo, mapped)
+
+    create_null(data["df_null"], v1, v2, v3, condo, mapped)
```

Comment:
interesting. I will investigate, most likely we can just replace with text that says "this report not available for versions before 21v2"

Reply:
Also, if checking the condo and mapped option, it returns division by zero error in the mismatch generation part. I am looking into this part.
<img width="780" alt="image" src="https://user-images.githubusercontent.com/44253950/178793874-e844b199-12ea-4277-9092-e62505dccd23.png">

Reply:
That's because the tables in branch 317 have no mapped condos, it's not an issue with the app. We could implement nicer error messaging here, feel free to open an issue for that and implement a change.

Reply:
ok, we can handle this exception a little bit to avoid throw this messy error message

Reply:
okay want to report back on this that the main branch now with updated condo error fixed. The app is back to normal.